kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/coordinator/transaction/TransactionMarkerChannelHandler.java

```diff
@@ -286,33 +301,52 @@ private CompletableFuture<ChannelHandlerContext> authenticate(final ChannelHandl
     }
 
     private CompletableFuture<ChannelHandlerContext> authenticateInternal(ChannelHandlerContext channel) {
+        CompletableFuture<AbstractResponse> result = new CompletableFuture<>();
+
         RequestHeader header = new RequestHeader(
                 ApiKeys.SASL_AUTHENTICATE,
                 ApiKeys.SASL_AUTHENTICATE.latestVersion(),
                 "tx", // ignored
                 correlationId.incrementAndGet()
         );
-        String prefix = "TX"; // the prefix TX means nothing, it is ignored by SaslUtils#parseSaslAuthBytes
-        String authUsername = transactionMarkerChannelManager.getAuthenticationUsername();
-        String authPassword = transactionMarkerChannelManager.getAuthenticationPassword();
-        String usernamePassword = prefix
-                + "\u0000" + authUsername
-                + "\u0000" + authPassword;
-        byte[] saslAuthBytes = usernamePassword.getBytes(UTF_8);
-        SaslAuthenticateRequest request = new SaslAuthenticateRequest
-                .Builder(ByteBuffer.wrap(saslAuthBytes))
-                .build();
+        Authentication authentication = transactionMarkerChannelManager.getAuthentication();
 
-        ByteBuffer buffer = request.serialize(header);
+        try {
+            byte[] saslAuthBytes;
+            String commandData = authentication.getAuthData().getCommandData();
+            if (authentication instanceof AuthenticationToken) {
+                String prefix = "TX"; // the prefix TX means nothing, it is ignored by SaslUtils#parseSaslAuthBytes
+                String authUsername = transactionMarkerChannelManager.getAuthenticationUsername();
+                String authPassword = "token:" + commandData;
+                String usernamePassword = prefix
+                        + "\u0000" + authUsername
+                        + "\u0000" + authPassword;
+                saslAuthBytes = usernamePassword.getBytes(UTF_8);
+            } else if (authentication instanceof AuthenticationOAuth2) {
+                saslAuthBytes = new OAuthBearerClientInitialResponse(commandData).toBytes();
+            } else {
+                log.error("Unknown authentication : {}", authentication);
+                saslAuthBytes = new byte[0];
+            }
```

Comment:
```java
            switch (mechanism) {
                case PlainSaslServer.PLAIN_MECHANISM:
                    // TODO:
                    break;
                case OAuthBearerLoginModule.OAUTHBEARER_MECHANISM:
                    // TODO:
                    break;
                default:
                    break;
            }
```

It's better to use the mechanism string to differ how to handle the `commandData` and other information (username and password).